src/wechaty_puppet_service/utils.py

```diff
@@ -58,7 +58,14 @@ def test_endpoint(end_point: str, log: Logger) -> int:
 
     res = True
     if port is None:
-        if ping(host) is False:
+        try:
+            if ping(host) is False:
+                res = False
+        except PermissionError as pe:
+            log.error(pe)
+            res = False
+        except errors.PingError as e:
+            log.error(f'got a ping error:\n{e}')
             res = False
```

Comment:
Your improvement look so great to me. But I don't in what situation the error will be raised.  Can add some more detailed test case for `PermissionError` and `PingError` to help us understand the code ?  If you can, it's great docs for developer to understand more about `endpoint`. How do you think about it ?

Reply:
Thanks for replay. :-)

There will be a `PermissionError`, when I set `end_point` to a service name and run script on a Linux machine without root. And without above codes, due to the outer try catch, I only got `Wechaty - ERROR - The network is not good, the bot will try to restart after 60 seconds.`, which make people confusing. There need more message for figuring out the true problem, so I add try catch for ping call and print possible exception message.

By the way, I optimized the comment.